File: backend-api/src/main/java/com/smartenergy/backendapi/model/Alerta.java
package com.smartenergy.backendapi.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name="Alerta")
public class Alerta extends BaseEntity{
    private String tipoAlerta;
    @Enumerated(EnumType.STRING)
    private EstadoAlerta estado;
    @ManyToOne
    @JoinColumn(name = "aula_id")
    private Aula aula;
    private LocalDateTime fechaHora;
}

File: mobile-app/app/src/main/java/com/example/smartenergy/model/Alerta.kt
package com.example.smartenergy.model

import java.time.LocalDateTime
import java.util.UUID

data class Alerta(
    val id: String = UUID.randomUUID().toString(),
    val tipoAlerta: String,
    val estado: EstadoAlerta,
    val aula: Aula? = null,
    val fechaHora: LocalDateTime = LocalDateTime.now()
) {
    val tipo: String get() = tipoAlerta
    val hora: String get() = fechaHora.format(java.time.format.DateTimeFormatter.ofPattern("h:mm a"))
}